kronjob/_kronjob.py

```diff
@@ -101,29 +78,62 @@ def serialize_k8s(k8s_object):
     )
 
 
-def build_k8s_object(aggregate_job):
+def build_k8s_object(aggregate_job, k8s_api_version=None, defaults=None):
+    k8s_api_version = k8s_api_version if k8s_api_version is not None else _K8S_API_VERSION
+    version_parts = str(k8s_api_version).split('.')
+    k8s_major, k8s_minor = int(version_parts[0]), int(version_parts[1])
+    if k8s_major != 1 or k8s_minor < 5:
+        raise ValueError('Unsupported kubernetes api version')
+    if k8s_minor >= 8:
+        cronjob_api_version = 'v1beta1'
+        CronJob = k8s_models.V1beta1CronJob
+        CronJobSpec = k8s_models.V1beta1CronJobSpec
+        JobTemplateSpec = k8s_models.V1beta1JobTemplateSpec
+    else:
+        cronjob_api_version = 'v2alpha1'
+        CronJob = k8s_models.V2alpha1CronJob
+        CronJobSpec = k8s_models.V2alpha1CronJobSpec
+        JobTemplateSpec = k8s_models.V2alpha1JobTemplateSpec
+
+    defaults = copy.deepcopy(defaults) if defaults is not None else {}
+    if 'containerName' not in defaults:
+        defaults['containerName'] = '{}-job'.format(aggregate_job['name'])
```

Comment:
should check to make sure this is under 63 chars in length, think kubernetes has a limit on container names, and there are other issues (see https://github.com/better/mortgage/pull/18124)

probably should make the max length configurable

Reply:
👍 I'll add a validation. I'll probably just hard code 63 chars for now unless the limit's likely to change for some reason?

Reply:
I added a validation on name since it has to be <=52 chars for cronjobs which will act as a proxy for this.